static/scripts/PlotHandler.js

```diff
@@ -116,9 +132,87 @@ class PlotWorker {
     }
     return;
   }
+
+  getPlotDataAsCSV(zip) {
+    let csvContent = "";
+    if (this.isSingleInput) {
+      csvContent += singlePlotCalculator.tranformToCSV(this.plotData);
+    } else {
+      csvContent += doublePlotCalculator.tranformToCSV(this.plotData);
+    }
+    zip.file(this.plotName + ".csv", csvContent);
+  }
+}
+
+class singleInputPlotCalculator {
+  constructor() {}
+
+  tranformToCSV(plotData) {
+    let labels = [];
+    plotData.forEach((dataset) => {
+      labels.push(dataset.label);
+    });
+    let heading = labels.join(";") + "\n";
+    let maxColumnLength = 0;
+    plotData.forEach((dataset) => {
+      if (dataset.data.length > maxColumnLength) {
+        maxColumnLength = dataset.data.length;
+      }
+    });
+    let rows = [];
+    for (let i = 0; i < maxColumnLength; ++i) {
+      let column = [];
+      plotData.forEach((dataset) => {
+        let data = dataset.data;
+        if (data.length > i) column.push(data[i].y);
+        else column.push("");
+      });
+      rows.push(column.join(";"));
+    }
+    let csvContent = heading + rows.join("\n");
+    return csvContent;
+  }
+}
+
+class doubleInputPlotCalculator {
+  constructor() {}
+  tranformToCSV(plotData) {
+    // not sure if this is a good way to do this
+    let labels = [];
+    plotData.forEach((dataset) => {
+      labels.push(dataset.label + "_x");
+      labels.push(dataset.label + "_y");
+    });
+    let heading = labels.join(";") + "\n";
+    let maxColumnLength = 0;
+    plotData.forEach((dataset) => {
+      if (dataset.data.length > maxColumnLength) {
+        maxColumnLength = dataset.data.length;
+      }
+    });
```

Comment:
You could use `Array.prototype.map` in combination with `Math.max` like this

`maxColumnLength = Math.max(...(plotData.map(dataset => dataset.data.length)))`